test/Microsoft.AspNet.Mvc.Core.Test/ControllerActionInvokerTest.cs

```diff
@@ -1203,6 +1203,647 @@ private struct SampleStruct
+
+            // Assert
+            Assert.Same(expected, context.Exception);
+            Assert.Same(expected, context.ExceptionDispatchInfo.SourceException);
+
+            resourceFilter.Verify(
+                f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()),
+                Times.Once());
+        }
+
+        [Fact]
+        public async Task InvokeAction_InvokesAsyncResourceFilter_HandleException_BySettingNull()
+        {
+            // Arrange
+            ResourceExecutedContext context = null;
+            var resourceFilter = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>(async (c, next) =>
+                {
+                    context = await next();
+
+                    Assert.Same(_actionException, context.Exception);
+                    Assert.Same(_actionException, context.ExceptionDispatchInfo.SourceException);
+
+                    context.Exception = null;
+                })
+                .Verifiable();
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter.Object }, actionThrows: true);
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            resourceFilter.Verify(
+                f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()),
+                Times.Once());
+        }
+
+        [Fact]
+        public async Task InvokeAction_InvokesAsyncResourceFilter_ThrowsUnhandledException()
+        {
+            // Arrange
+            var expected = new DataMisalignedException();
+
+            ResourceExecutedContext context = null;
+            var resourceFilter1 = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter1
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>(async (c, next) =>
+                {
+                    context = await next();
+                })
+                .Verifiable();
+
+            var resourceFilter2 = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter2
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>((c, next) =>
+                {
+                    throw expected;
+                })
+                .Verifiable();
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter1.Object, resourceFilter2.Object }, actionThrows: true);
```

Comment:
Doesn't much matter here. I think generally these tests have the action throw when we don't expect it to run.